### ISSUE TITLE
rsem-prepare-reference with STAR using Trusted Sources

### ISSUE THREAD

**Issue by deto** [deweylab/RSEM#49]
I ran rsem-prepare-reference in '-star' mode but also using the '--trusted-sources' to limit the source to "BestRefSeq" and "Curated Genomic".  However, when running rsem-calculate-expression, after STAR alignment, there is an error that the BAM file contains many more sequences than RSEM knows.

It looks like the issue is that the --trusted-sources information is not fed into STAR when it creates it's reference, and so it uses many more transcripts than RSEM is keeping in its own files.  

I think I'll just try a workaround where I pre-filter my GTF file, but I just thought I'd bring this to your attention so you could either fix or have the tool raise an error message when -star and --trusted-sources are used together.

**Comment by jhl667** [comment]
For the record, I ran in to this same issue.

**Comment by rocanja** [comment]
Encountered the same problem! Seeing that this issue has been open since 2017, it would be nice to at least have a note about this problem in the vignette, please!

**Comment by MattMyint** [comment]
Just adding info for others who may encounter this too.

> I think I'll just try a workaround where I pre-filter my GTF file, but I just thought I'd bring this to your attention so you could either fix or have the tool raise an error message when -star and --trusted-sources are used together.

What worked for me was the following:
```
# Removing space in "Curated Genomic" because it was giving me a headache
sed -e 's/Curated Genomic/CuratedGenomic/g' GCF_000001405.26_GRCh38_genomic.gff > GCF_000001405.26_GRCh38_genomic.nws.gff

# Filtering the gff file to only include my selected sources
awk -F'\t' 'NR<=7; NR>7 { if ($2 == "BestRefSeq" || $2 == "CuratedGenomic") print $0 }' GCF_000001405.26_GRCh38_genomic.nws.gff > GCF_000001405.26_GRCh38_genomic.BRS-CG.gff 

rsem-prepare-reference --gff3 GCF_000001405.26_GRCh38_genomic.BRS-CG.gff --star -p 60 GCF_000001405.26_GRCh38_genomic.primary_assembly.fna rsem-hg38

# all this is done in a directory that holds my reference genome files, which I will later point rsem-calculate-expression in STAR mode to
```

tl;dr reporting here that pre-filtering the gtf and using that to prepare the rsem reference worked for me
